control-plane/agents/core/src/core/scheduling/mod.rs

```diff
@@ -68,6 +68,35 @@ impl NodeFilters {
             .into_iter()
             .any(|node_spec| node_spec.id() == &item.pool.node)
     }
+
+    /// Should only attempt to use online nodes
+    pub(crate) fn online(_request: &GetSuitableNodesContext, item: &NodeItem) -> bool {
+        item.node_wrapper.is_online()
+    }
+
+    /// Should abide by the topology constraints
+    pub(crate) fn topology(request: &GetSuitableNodesContext, item: &NodeItem) -> bool {
```

Comment:
We shouldn't add this, the topology is for data replicas and not nexuses. And until we decide what to do with it we shouldn't be adding more code using it.

Reply:
Removing it for now as we are not honoring any topology. Need to revisit this once we have the changes.